packages/agent-toolkit/src/core/tools/platform-api-tools/get-board-items-page-tool/get-board-items-page-tool.ts

```diff
@@ -68,6 +73,19 @@ export class GetBoardItemsPageTool extends BaseMondayApiTool<GetBoardItemsPageTo
   }
   
   protected async executeInternal(input: ToolInputType<GetBoardItemsPageToolInput>): Promise<ToolOutputType<never>> {
+    // Passing filters + cursor returns an error as cursor has them encoded in it
+    const canIncludeFilters = !input.cursor;
+
+    if(canIncludeFilters && input.searchTerm) {
+      input.itemIds = await this.getItemIdsFromSmartSearchAsync(input);
```

Comment:
how about the following instead to avoid modifying the `input` object
```
let itemIds = input.itemIds;

if (...) {
    itemIds = await this.getItemIds....
    ....
}
```